src/main/java/org/sonarlint/intellij/util/SonarLintAppUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.util;
+
+import com.intellij.openapi.components.ApplicationComponent;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+public class SonarLintAppUtils extends ApplicationComponent.Adapter {
+  protected SonarLintAppUtils() {
+    super();
+  }
+
+  public boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nullable Module module) {
+    return SonarLintUtils.shouldAnalyzeAutomatically(file, module);
+  }
+
+  public boolean shouldAnalyze(@Nullable VirtualFile file, @Nullable Module module) {
```

Comment:
Who would call this with a `null` file? The question "should analyze?" makes no sense if there is no file to analyze.